projects/igniteui-angular/src/lib/data-operations/data-util.ts

```diff
@@ -165,10 +165,12 @@ export class DataUtil {
 
     public static getHierarchy(gRow: IGroupByRecord): Array<IGroupByKey> {
         const hierarchy: Array<IGroupByKey> = [];
-        hierarchy.push({ fieldName: gRow.expression.fieldName, value: gRow.value });
-        while (gRow.groupParent) {
-            gRow = gRow.groupParent;
-            hierarchy.unshift({ fieldName: gRow.expression.fieldName, value: gRow.value });
+        if (gRow !== undefined) {
+            hierarchy.push({ fieldName: gRow.expression.fieldName, value: gRow.value });
```

Comment:
Is there any chance to have no expression and get in trouble getting the fieldName?

Reply:
I added this check because I ran into a few errors where we would get undefined gRow argument, but I never saw one without expressions, so I don't think it is possible.